### PR TITLE
upgrade to polkadot-v1.6.0

### LINKED ISSUE #264: upgrade to polkadot-v1.6
noteworthy for us
* this will enable XCM teleport of TEER to asset hub with: https://github.com/paritytech/polkadot-sdk/pull/2388 (avoiding the temporary hack in trappist, where they use withdrawTeleport pallet
* better storage version init for migrations: https://github.com/paritytech/polkadot-sdk/pull/1297/
* [x] needs migration of xcmp-queue

needs https://github.com/integritee-network/pallets/issues/250

we skip v1.4. and v1.5 because ORML skipped them as well from 0.6.7 to 0.7.0
we don't go directly to 1.7 [system chains will hop from 1.3 to 1.7](https://github.com/polkadot-fellows/runtimes/issues/140) but that's WIP as of today
* 1.7 will introduce XCMv4. better to have a working feature-complete setup with XCMv3 as there may be laggard siblings

### PR DESCRIPTION
**Pull request by brenzi**
closes #264 

* [x] needs migration of xcmp-queue
* [x] rerun benchmarks
* [x] try-runtime
* [x] chopsticks xcm with Karura
* [x] re-genesis rococo to fix broken chain???

### CODE REVIEW

**Comment by brenzi** [comment]
currently stuck here: 
https://substrate.stackexchange.com/questions/10986/runtime-upgrade-for-parachainsystemhostconfiguration

File: polkadot-parachains/integritee-runtime/src/lib.rs [hunk 1]
```diff
@@ -451,24 +449,58 @@ impl pallet_preimage::Config for Runtime {
 parameter_types! {
 	pub const ReservedXcmpWeight: Weight = MAXIMUM_BLOCK_WEIGHT.saturating_div(4);
 	pub const ReservedDmpWeight: Weight = MAXIMUM_BLOCK_WEIGHT.saturating_div(4);
+	pub const RelayOrigin: AggregateMessageOrigin = AggregateMessageOrigin::Parent;
 }
 
 impl cumulus_pallet_parachain_system::Config for Runtime {
+	type WeightInfo = ();
```
Comment (clangenb) [inline polkadot-parachains/integritee-runtime/src/lib.rs:456]: Oh, a todo I guess.

File: polkadot-parachains/integritee-runtime/src/lib.rs [hunk 2]
```diff
@@ -861,10 +893,6 @@ mod benches {
 		[pallet_child_bounties, ChildBounties]
 		[pallet_claims, Claims]
 		[pallet_collective, Council]
-		// Fixme #177: For some reason the benchmarks write both to the same pallet_collective weight
-		// file. No reason to run both before we fix that, as the pallet_collective benchmark
-		// takes the longest.
-		// [pallet_collective, TechnicalCommittee]
 		[pallet_democracy, Democracy]
```
Comment (clangenb) [inline polkadot-parachains/integritee-runtime/src/lib.rs:896]: what is the resolution here?

File: polkadot-parachains/integritee-runtime/src/weights/pallet_xcm.rs [hunk 3]
```diff
@@ -44,42 +62,106 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
...
 	}
 	/// Storage: `Benchmark::Override` (r:0 w:0)
 	/// Proof: `Benchmark::Override` (`max_values`: None, `max_size`: None, mode: `Measured`)
 	fn execute() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		Weight::from_parts(10_240_000, 0)
+		// Minimum execution time: 18_446_744_073_709_551_000 picoseconds.
+		Weight::from_parts(18_446_744_073_709_551_000, 0)
```
Comment (clangenb) [inline polkadot-parachains/integritee-runtime/src/weights/pallet_xcm.rs:164]: We have again this crazy weight here.

File: polkadot-parachains/src/chain_spec.rs [hunk 4]
```diff
@@ -142,21 +143,35 @@ pub fn shell_chain_spec(
...
+	.with_protocol_id(relay_chain.protocol_id())
+	.with_chain_type(relay_chain.chain_type())
+	.with_properties(properties)
+	.with_genesis_config_patch(shell_genesis_config(
+		root.clone(),
+		endowed.clone(),
+		authorities.clone(),
+		para_id,
+	))
+	.build()
```
Comment (clangenb) [inline polkadot-parachains/src/chain_spec.rs:162]: 💯

File: polkadot-parachains/src/chain_spec.rs [hunk 5]
```diff
@@ -169,119 +184,78 @@ pub fn integritee_chain_spec(
...
+			"allowSkippingAttestation": true,
 		},
-		claims: Default::default(),
-		treasury: Default::default(),
-		transaction_payment: Default::default(),
-	}
+		"council": {
+			"members": vec![root_key.clone()]
+		},
+		"technicalcommittee": {
```
Comment (clangenb) [inline polkadot-parachains/src/chain_spec.rs:232]: hmm, why is this suddenly like this. It seems less typesafe too me?

File: polkadot-parachains/src/chain_spec.rs [hunk 6]
```diff
@@ -296,34 +270,32 @@ pub enum RelayChain {
 	Moonbase,
 }
 
-pub fn shell_rococo_config() -> Result<ShellChainSpec, String> {
-	ShellChainSpec::from_json_bytes(&include_bytes!("../chain-specs/integritee-rococo.json")[..])
+pub fn shell_rococo_config() -> Result<ChainSpec, String> {
+	ChainSpec::from_json_bytes(&include_bytes!("../chain-specs/integritee-rococo.json")[..])
```
Comment (clangenb) [inline polkadot-parachains/src/chain_spec.rs:274]: Ah, I see it is less typesafe because there is a generic chainspec here.

**Review by clangenb — COMMENTED** [review]
Minor questions, otherwise it looks good.

File: polkadot-parachains/integritee-runtime/src/lib.rs (hunk 2 above)
Comment (brenzi) [reply polkadot-parachains/integritee-runtime/src/lib.rs:896]: I see no problem which needs solving. both instances can take their weights from the same benchmark. its the same pallet so it's the same weights...

File: polkadot-parachains/integritee-runtime/src/weights/pallet_xcm.rs (hunk 3 above)
Comment (brenzi) [reply polkadot-parachains/integritee-runtime/src/weights/pallet_xcm.rs:164]: will sort that

File: polkadot-parachains/src/chain_spec.rs (hunk 5 above)
Comment (brenzi) [reply polkadot-parachains/src/chain_spec.rs:232]: good catch
the goal is to no longer build native runtimes, so all needs to be defined in json

**Pull request merged**